workerpool/example/tasks.go

```diff
@@ -1,19 +1,42 @@
 package main
 
-import "time"
+import (
+	"errors"
+	"math/rand"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/sean-/patterns/workerpool"
+)
 
 type taskReal struct {
+	log zerolog.Logger
+
+	tid      workerpool.ThreadID
 	finishFn func()
 }
 
-func (rt *taskReal) DoWork() {
+func (rt *taskReal) DoWork() error {
 	time.Sleep(2 * time.Second)
+
+	if rand.Intn(5) == 1 {
```

Comment:
Turn `5` into a named constant.  Since this is a user/admin value, it might be worth expressing this as a pct, like `20%`, or `realWorkErrRate=0.2`.